test_033_parseTime_auto_month fails on 1.9.0

The mentioned test is failing for us on with metar 1.9.0. The environment is.

> platform linux -- Python 3.9.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0


```pytb
________________________ test_033_parseTime_auto_month _________________________

self = <metar.Metar.Metar object at 0x7ffff65a8370>, metarcode = 'KEWR 291651Z'
month = None, year = None, utcdelta = None, strict = True

    def __init__(self, metarcode, month=None, year=None, utcdelta=None, strict=True):
        """
        Parse raw METAR code.
    
        Parameters
        ----------
        metarcode : str
        month, year : int, optional
          Date values to be used when parsing a non-current METAR code. If not
          provided, then the month and year are guessed from the current date.
        utcdelta : int or datetime.timedelta, optional
          An int of hours or a timedelta object used to specify the timezone.
        strict : bool (default is True)
          This option determines if a ``ParserError`` is raised when
          unparsable groups are found or an unexpected exception is encountered.
          Setting this to `False` will prevent exceptions from being raised and
          only generate warning messages.
        """
    
        self.code = metarcode  # original METAR code
        self.type = "METAR"  # METAR (routine) or SPECI (special)
        self.correction = None  # COR (corrected - WMO spec)
        self.mod = "AUTO"  # AUTO (automatic) or COR (corrected - US spec)
        self.station_id = None  # 4-character ICAO station code
        self.time = None  # observation time [datetime]
        self.cycle = None  # observation cycle (0-23) [int]
        self.wind_dir = None  # wind direction [direction]
        self.wind_speed = None  # wind speed [speed]
        self.wind_gust = None  # wind gust speed [speed]
        self.wind_dir_from = None  # beginning of range for win dir [direction]
        self.wind_dir_to = None  # end of range for wind dir [direction]
        self.vis = None  # visibility [distance]
        self.vis_dir = None  # visibility direction [direction]
        self.max_vis = None  # visibility [distance]
        self.max_vis_dir = None  # visibility direction [direction]
        self.temp = None  # temperature (C) [temperature]
        self.dewpt = None  # dew point (C) [temperature]
        self.press = None  # barometric pressure [pressure]
        self.runway = []  # runway visibility (list of tuples)
        self.weather = []  # present weather (list of tuples)
        self.recent = []  # recent weather (list of tuples)
        self.sky = []  # sky conditions (list of tuples)
        self.windshear = []  # runways w/ wind shear (list of strings)
        self.wind_speed_peak = None  # peak wind speed in last hour
        self.wind_dir_peak = None  # direction of peak wind speed in last hour
        self.peak_wind_time = None  # time of peak wind observation [datetime]
        self.wind_shift_time = None  # time of wind shift [datetime]
        self.max_temp_6hr = None  # max temp in last 6 hours
        self.min_temp_6hr = None  # min temp in last 6 hours
        self.max_temp_24hr = None  # max temp in last 24 hours
        self.min_temp_24hr = None  # min temp in last 24 hours
        self.press_sea_level = None  # sea-level pressure
        self.precip_1hr = None  # precipitation over the last hour
        self.precip_3hr = None  # precipitation over the last 3 hours
        self.precip_6hr = None  # precipitation over the last 6 hours
        self.precip_24hr = None  # precipitation over the last 24 hours
        self.snowdepth = None  # snow depth (distance)
        self.ice_accretion_1hr = None  # ice accretion over the past hour
        self.ice_accretion_3hr = None  # ice accretion over the past 3 hours
        self.ice_accretion_6hr = None  # ice accretion over the past 6 hours
        self._trend = False  # trend groups present (bool)
        self._trend_groups = []  # trend forecast groups
        self._remarks = []  # remarks (list of strings)
        self._unparsed_groups = []
        self._unparsed_remarks = []
    
        self._now = datetime.datetime.utcnow()
        if utcdelta:
            self._utcdelta = utcdelta
        else:
            self._utcdelta = datetime.datetime.now() - self._now
    
        self._month = month
        self._year = year
    
        # Do some string prep before parsing
        code = _sanitize(self.code)
        try:
            ngroup = len(self.handlers)
            igroup = 0
            ifailed = -1
            while igroup < ngroup and code:
                pattern, handler, repeatable = self.handlers[igroup]
                if debug:
                    _logger.debug("%s: %s", handler.__name__, code)
                m = pattern.match(code)
                while m:
                    ifailed = -1
                    if debug:
                        _report_match(handler, m.group())
>                   handler(self, m.groupdict())

metar/Metar.py:444: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <metar.Metar.Metar object at 0x7ffff65a8370>
d = {'day': '29', 'hour': '16', 'min': '51'}

    def _handleTime(self, d):
        """
        Parse the observation-time group.
    
        The following attributes are set:
            time   [datetime]
            cycle  [int]
            _day   [int]
            _hour  [int]
            _min   [int]
        """
        self._day = int(d["day"])
        if not self._month:
            self._month = self._now.month
            if self._day > self._now.day:
                if self._month == 1:
                    self._month = 12
                else:
                    self._month = self._month - 1
        if not self._year:
            self._year = self._now.year
            if self._month > self._now.month:
                self._year = self._year - 1
            elif self._month == self._now.month and self._day > self._now.day:
                self._year = self._year - 1
        self._hour = int(d["hour"])
        self._min = int(d["min"])
>       self.time = datetime.datetime(
            self._year, self._month, self._day, self._hour, self._min
        )
E       ValueError: day is out of range for month

metar/Metar.py:598: ValueError

During handling of the above exception, another exception occurred:

    def test_033_parseTime_auto_month():
        """Test assignment of time to previous month."""
        next_day = tomorrow.day
        if next_day > today.day:
            last_month = ((today.month - 2) % 12) + 1
            last_year = today.year - 1
    
            timestr = "%02d1651Z" % (next_day)
>           report = Metar.Metar("KEWR " + timestr)

test/test_metar.py:208: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <metar.Metar.Metar object at 0x7ffff65a8370>, metarcode = 'KEWR 291651Z'
month = None, year = None, utcdelta = None, strict = True

    def __init__(self, metarcode, month=None, year=None, utcdelta=None, strict=True):
        """
        Parse raw METAR code.
    
        Parameters
        ----------
        metarcode : str
        month, year : int, optional
          Date values to be used when parsing a non-current METAR code. If not
          provided, then the month and year are guessed from the current date.
        utcdelta : int or datetime.timedelta, optional
          An int of hours or a timedelta object used to specify the timezone.
        strict : bool (default is True)
          This option determines if a ``ParserError`` is raised when
          unparsable groups are found or an unexpected exception is encountered.
          Setting this to `False` will prevent exceptions from being raised and
          only generate warning messages.
        """
    
        self.code = metarcode  # original METAR code
        self.type = "METAR"  # METAR (routine) or SPECI (special)
        self.correction = None  # COR (corrected - WMO spec)
        self.mod = "AUTO"  # AUTO (automatic) or COR (corrected - US spec)
        self.station_id = None  # 4-character ICAO station code
        self.time = None  # observation time [datetime]
        self.cycle = None  # observation cycle (0-23) [int]
        self.wind_dir = None  # wind direction [direction]
        self.wind_speed = None  # wind speed [speed]
        self.wind_gust = None  # wind gust speed [speed]
        self.wind_dir_from = None  # beginning of range for win dir [direction]
        self.wind_dir_to = None  # end of range for wind dir [direction]
        self.vis = None  # visibility [distance]
        self.vis_dir = None  # visibility direction [direction]
        self.max_vis = None  # visibility [distance]
        self.max_vis_dir = None  # visibility direction [direction]
        self.temp = None  # temperature (C) [temperature]
        self.dewpt = None  # dew point (C) [temperature]
        self.press = None  # barometric pressure [pressure]
        self.runway = []  # runway visibility (list of tuples)
        self.weather = []  # present weather (list of tuples)
        self.recent = []  # recent weather (list of tuples)
        self.sky = []  # sky conditions (list of tuples)
        self.windshear = []  # runways w/ wind shear (list of strings)
        self.wind_speed_peak = None  # peak wind speed in last hour
        self.wind_dir_peak = None  # direction of peak wind speed in last hour
        self.peak_wind_time = None  # time of peak wind observation [datetime]
        self.wind_shift_time = None  # time of wind shift [datetime]
        self.max_temp_6hr = None  # max temp in last 6 hours
        self.min_temp_6hr = None  # min temp in last 6 hours
        self.max_temp_24hr = None  # max temp in last 24 hours
        self.min_temp_24hr = None  # min temp in last 24 hours
        self.press_sea_level = None  # sea-level pressure
        self.precip_1hr = None  # precipitation over the last hour
        self.precip_3hr = None  # precipitation over the last 3 hours
        self.precip_6hr = None  # precipitation over the last 6 hours
        self.precip_24hr = None  # precipitation over the last 24 hours
        self.snowdepth = None  # snow depth (distance)
        self.ice_accretion_1hr = None  # ice accretion over the past hour
        self.ice_accretion_3hr = None  # ice accretion over the past 3 hours
        self.ice_accretion_6hr = None  # ice accretion over the past 6 hours
        self._trend = False  # trend groups present (bool)
        self._trend_groups = []  # trend forecast groups
        self._remarks = []  # remarks (list of strings)
        self._unparsed_groups = []
        self._unparsed_remarks = []
    
        self._now = datetime.datetime.utcnow()
        if utcdelta:
            self._utcdelta = utcdelta
        else:
            self._utcdelta = datetime.datetime.now() - self._now
    
        self._month = month
        self._year = year
    
        # Do some string prep before parsing
        code = _sanitize(self.code)
        try:
            ngroup = len(self.handlers)
            igroup = 0
            ifailed = -1
            while igroup < ngroup and code:
                pattern, handler, repeatable = self.handlers[igroup]
                if debug:
                    _logger.debug("%s: %s", handler.__name__, code)
                m = pattern.match(code)
                while m:
                    ifailed = -1
                    if debug:
                        _report_match(handler, m.group())
                    handler(self, m.groupdict())
                    code = code[m.end():]
                    if self._trend:
                        code = self._do_trend_handlers(code)
                    if not repeatable:
                        break
    
                    if debug:
                        _logger.debug("%s: %s", handler.__name__, code)
                    m = pattern.match(code)
                if not m and ifailed < 0:
                    ifailed = igroup
                igroup += 1
                if igroup == ngroup and not m:
                    pattern, handler = (UNPARSED_RE, _unparsedGroup)
                    if debug:
                        _logger.debug("%s: %s", handler.__name__, code)
                    m = pattern.match(code)
                    if debug:
                        _report_match(handler, m.group())
                    handler(self, m.groupdict())
                    code = code[m.end():]
                    igroup = ifailed
                    ifailed = -2  # if it's still -2 when we run out of main-body
                    #  groups, we'll try parsing this group as a remark
            if pattern == REMARK_RE or self.press:
                while code:
                    for pattern, handler in self.remark_handlers:
                        if debug:
                            _logger.debug("%s: %s", handler.__name__, code)
                        m = pattern.match(code)
                        if m:
                            if debug:
                                _report_match(handler, m.group())
                            handler(self, m.groupdict())
                            code = pattern.sub("", code, 1)
                            break
    
        except Exception as err:
            message = ("%s failed while processing '%s'\n\t%s") % (
                handler.__name__,
                code,
                "\n\t".join(err.args),
            )
            if strict:
>               raise ParserError(message)
E               metar.Metar.ParserError: _handleTime failed while processing '291651Z '
E                       day is out of range for month

metar/Metar.py:489: ParserError
```

Fails today again with #166 applied.

```python
________________________ test_033_parseTime_auto_month _________________________

    def test_033_parseTime_auto_month():
        """Test assignment of time to previous month."""
        # To test this, we need a date last month that is with a greater day
        # than today, which does not always exist!
        last_month = today.replace(day=1) - timedelta(days=1)
        if last_month.day >= today.day:
            return
    
        report = Metar.Metar(f"KEWR {last_month.day:02.0f}1651Z")
        assert report.decode_completed
        assert report.time.day == last_month.day
>       assert report.time.month == last_month.month
E       assert 3 == 2
E        +  where 3 = datetime.datetime(2022, 3, 28, 16, 51).month
E        +    where datetime.datetime(2022, 3, 28, 16, 51) = <metar.Metar.Metar object at 0x7ffff6549fa0>.time
E        +  and   2 = datetime.datetime(2022, 2, 28, 3, 57, 8, 516562).month

test/test_metar.py:211: AssertionError
```

--- a/test/test_metar.py
+++ b/test/test_metar.py
@@ -199,20 +199,17 @@ def test_032_parseTime_specify_month():
 
 def test_033_parseTime_auto_month():
     """Test assignment of time to previous month."""
-    next_day = tomorrow.day
-    if next_day > today.day:
-        last_month = ((today.month - 2) % 12) + 1
-        last_year = today.year - 1
+    # To test this, we need a date last month that is with a greater day
+    # than today, which does not always exist!
+    last_month = today.replace(day=1) - timedelta(days=1)
+    if last_month.day <= today.day:
+        return
 
-        timestr = "%02d1651Z" % (next_day)
-        report = Metar.Metar("KEWR " + timestr)
-        assert report.decode_completed
-        assert report.time.day == next_day
-        assert report.time.month == last_month
-        if today.month > 1:
-            assert report.time.year == today.year
-        else:
-            assert report.time.year == last_year
+    report = Metar.Metar(f"KEWR {last_month.day:02.0f}1651Z")
+    assert report.decode_completed
+    assert report.time.day == last_month.day
+    assert report.time.month == last_month.month
+    assert report.time.year == last_month.year
 
 
 def test_034_parseTime_auto_year():

